BindingContext/BPoP.md

```diff
@@ -0,0 +1,334 @@
+
+The browser attaches BPoP proofs to a request if there exists a config that either:
+
+- Has an exact match between the BPoP origin and the canonicalized host of the retrieval's URI and the `subdomains` flag is false
+- Has a domain match between the BPoP origin and the canonicalized host of the retrieval's URI and the `subdomains` flag is true
+
+The semantics of `SameSite` match the cookie attributes. That is, if the browser would not attach a cookie with SameSite=Lax to a request, and the server has initialized BPoP with SameSite=Lax, the browser should not attach a BPoP proof to the request.
+
+TBD: Add how `browser-policy` is defined and how it will impact the key management.
+
+### Application model
+
+#### BPoP key verification
+
+The application generated cookie bound by BPoP should not need to verify the asymmetric key everytime. Instead, the application should cache the `key-verification` status and should be able to request the key verification only when the cache is expired or invalidated. Cache validity can be equivalent to `BPoP-Nonce` validity.
+
+The browser initiates BPoP proof of possession only if:
+
+- BPoP nonce is expired or invalidated; set cache expiration time to `BPoP-Nonce` expiration time.
+
+```
+BPoP: enabled, expires-in=300
+```
+
+#### BPoP background refresh
+
+The server response should also include a `refresh-in` parameter, that enables the app to refresh the nonce in a certain interval. This should help the APIs dependent on BPoP to proactively refresh the nonce in the background avoiding the need to refresh the nonce on demand.
+
+```
+BPoP: enabled, refresh-in=60
+```
+
+### Server challenge
+
+A server may reject a BPoP proof because its nonce is missing or out of date:
+
+```
+HTTP/1.1 401 Unauthorized
+WWW-Authenticate: BPoP error="use_bpop_nonce", error_description="Web server requires nonce in BPoP proof"
+BPoP-Nonce: eyJ7S_zG.eyJH0-Z.HX4w-7v
+```
+
+In such a case, the browser should update the BPoP proof and retry the request immediately. Servers SHALL not ask for more than one retry this way.
+
+Such retries are intended to be seen by the client as part of a single HTTP fetch. That is, update [Fetch section 4.3 HTTP Fetch](https://fetch.spec.whatwg.org/#http-fetch) with additional steps. If _response_ is a 401 status code, and the response includes a `WWW-Authentication` header indicating the `BPoP` scheme with `error` equal to `use_bpop_nonce` and the response includes a `BPoP-Nonce` header, store the nonce, regenerate the BPoP Proof, and set the _actualResponse_ to the result of running HTTP-network-or-cache fetch for the updated _fetchParams_.
+
+### Server update
+
+A server may also return a new BPoP nonce on any 200 response.
+
+```
+HTTP/1.1 200 Ok
+BPoP-Nonce: eyJ7S_zG.eyJH0-Z.HX4w-7v
+refresh-in: 300
+expires-in: 300
+```
+
+The client SHOULD start using the new nonce on the next request.
```

Comment:
Is the server expected to provide rotation semantics for the nonce by supporting multiple nonces with overlapping windows? If so, should the server start sending a new BPoP nonce back even while it continues to accept a previous one for some period of time?